From 55a1e7f8a476c5e49ccdeeb6284e2a550781e5bf Mon Sep 17 00:00:00 2001
From: Carl Smedstad <carl.smedstad@protonmail.com>
Date: Sat, 30 Mar 2024 11:47:48 +0100
Subject: [PATCH] test: Replace dependency pytest-lazy-fixture

A new release of the pytest-lazy-fixture has not been published since
February 2020 (v0.6.3) and it is not compatible with recent pytest
versions, starting from v8.0.0 (I think).

Replace it with pytest-lazy-fixtures (note the 's' at the end) which is
actively maintained, see:

  https://github.com/dev-petrov/pytest-lazy-fixtures
---
 pdm.lock                                      | 27 +++++++------------
 pyproject.toml                                |  3 +--
 tests/conftest.py                             |  6 ++---
 tests/unit/test_events.py                     |  4 +--
 .../test_websocket_handlers/test_listeners.py |  8 +++---
 5 files changed, 19 insertions(+), 29 deletions(-)

diff --git a/pdm.lock b/pdm.lock
index dc29baa97..19f862c99 100644
--- a/pdm.lock
+++ b/pdm.lock
@@ -5,7 +5,7 @@
 groups = ["default", "standard", "jwt", "pydantic", "cli", "picologging", "dev-contrib", "piccolo", "prometheus", "dev", "mako", "test", "brotli", "cryptography", "linting", "attrs", "opentelemetry", "docs", "redis", "sqlalchemy", "full", "annotated-types", "jinja", "structlog", "minijinja"]
 strategy = ["cross_platform"]
 lock_version = "4.4.1"
-content_hash = "sha256:b8bd4e90f7c2d49ae6f87b84fb7cc6dc32654ef014fe8393d158b51f73f520d7"
+content_hash = "sha256:02bff62494e760a50a2236bd0c520d6cd1dba6ff8b35570835f5761f1c9b6db0"
 
 [[package]]
 name = "accessible-pygments"
@@ -993,7 +993,7 @@ files = [
 name = "domdf-python-tools"
 version = "3.8.0.post2"
 requires_python = ">=3.6"
-summary = "Helpful functions for Python 🐍 🛠️"
+summary = "Helpful functions for Python 🐍 🛠️"
 dependencies = [
     "importlib-metadata>=3.6.0; python_version < \"3.9\"",
     "natsort>=7.0.1",
@@ -2867,15 +2867,16 @@ files = [
 ]
 
 [[package]]
-name = "pytest-lazy-fixture"
-version = "0.6.3"
-summary = "It helps to use fixtures in pytest.mark.parametrize"
+name = "pytest-lazy-fixtures"
+version = "1.0.7"
+requires_python = ">=3.8,<4.0"
+summary = "Allows you to use fixtures in @pytest.mark.parametrize."
 dependencies = [
-    "pytest>=3.2.5",
+    "pytest>=7",
 ]
 files = [
-    {file = "pytest-lazy-fixture-0.6.3.tar.gz", hash = "sha256:0e7d0c7f74ba33e6e80905e9bfd81f9d15ef9a790de97993e34213deb5ad10ac"},
-    {file = "pytest_lazy_fixture-0.6.3-py3-none-any.whl", hash = "sha256:e0b379f38299ff27a653f03eaa69b08a6fd4484e46fd1c9907d984b9f9daeda6"},
+    {file = "pytest_lazy_fixtures-1.0.7-py3-none-any.whl", hash = "sha256:71428385b166cdb0fa3f2a3c72b5a5f95fa58f977816877b9b9c1e857143e194"},
+    {file = "pytest_lazy_fixtures-1.0.7.tar.gz", hash = "sha256:87ef7424dc0229ff9cb72d482f49b7806535c3500641f612c13ddf243c9adacb"},
 ]
 
 [[package]]
@@ -4000,16 +4001,6 @@ files = [
     {file = "types_pyOpenSSL-24.0.0.20240130-py3-none-any.whl", hash = "sha256:24a255458b5b8a7fca8139cf56f2a8ad5a4f1a5f711b73a5bb9cb50dc688fab5"},
 ]
 
-[[package]]
-name = "types-pytest-lazy-fixture"
-version = "0.6.3.20240310"
-requires_python = ">=3.8"
-summary = "Typing stubs for pytest-lazy-fixture"
-files = [
-    {file = "types-pytest-lazy-fixture-0.6.3.20240310.tar.gz", hash = "sha256:990bb96f556369ce8e37562171e75ad52956444b5a71c85b3e7f51db4cafcc7b"},
-    {file = "types_pytest_lazy_fixture-0.6.3.20240310-py3-none-any.whl", hash = "sha256:50618fba86e576e72cb3c200d940ad6d2a899c3493239806b907a964fbb527c6"},
-]
-
 [[package]]
 name = "types-python-jose"
 version = "3.3.4.20240106"
diff --git a/pyproject.toml b/pyproject.toml
index 3eb1d2489..33d0b2bb3 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -150,7 +150,6 @@ linting = [
   "pyright==1.1.344",
   "asyncpg-stubs",
   "types-beautifulsoup4",
-  "types-pytest-lazy-fixture",
   "types-python-jose",
   "types-pyyaml",
   "types-redis",
@@ -161,7 +160,7 @@ test = [
   "pytest<8.0.0",
   "pytest-asyncio",
   "pytest-cov",
-  "pytest-lazy-fixture",
+  "pytest-lazy-fixtures",
   "pytest-mock",
   "pytest-rerunfailures",
   "pytest-timeout",
diff --git a/tests/conftest.py b/tests/conftest.py
index e8007dea3..ca8df23b0 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -13,7 +13,7 @@ from typing import TYPE_CHECKING, Any, AsyncGenerator, Callable, Generator, Type
 from unittest.mock import AsyncMock, MagicMock
 
 import pytest
-from pytest_lazyfixture import lazy_fixture
+from pytest_lazy_fixtures import lf
 from redis.asyncio import Redis as AsyncRedis
 from redis.client import Redis
 from time_machine import travel
@@ -109,8 +109,8 @@ def cookie_session_backend(cookie_session_backend_config: CookieBackendConfig) -
 
 @pytest.fixture(
     params=[
-        pytest.param(lazy_fixture("cookie_session_backend_config"), id="cookie"),
-        pytest.param(lazy_fixture("server_side_session_config"), id="server-side"),
+        pytest.param(lf("cookie_session_backend_config"), id="cookie"),
+        pytest.param(lf("server_side_session_config"), id="server-side"),
     ]
 )
 def session_backend_config(request: pytest.FixtureRequest) -> ServerSideSessionConfig | CookieBackendConfig:
diff --git a/tests/unit/test_events.py b/tests/unit/test_events.py
index 082543b25..8245244a0 100644
--- a/tests/unit/test_events.py
+++ b/tests/unit/test_events.py
@@ -2,7 +2,7 @@ from typing import Any
 from unittest.mock import MagicMock
 
 import pytest
-from pytest_lazyfixture import lazy_fixture
+from pytest_lazy_fixtures import lf
 
 from litestar import Litestar, Request, get
 from litestar.events.emitter import SimpleEventEmitter
@@ -36,7 +36,7 @@ def async_listener(mock: MagicMock) -> EventListener:
     return listener_fn
 
 
-@pytest.mark.parametrize("event_listener", [lazy_fixture("sync_listener"), lazy_fixture("async_listener")])
+@pytest.mark.parametrize("event_listener", [lf("sync_listener"), lf("async_listener")])
 def test_event_listener(mock: MagicMock, event_listener: EventListener, anyio_backend: AnyIOBackend) -> None:
     @get("/")
     def route_handler(request: Request[Any, Any, Any]) -> None:
diff --git a/tests/unit/test_handlers/test_websocket_handlers/test_listeners.py b/tests/unit/test_handlers/test_websocket_handlers/test_listeners.py
index b42f41b80..08c74690d 100644
--- a/tests/unit/test_handlers/test_websocket_handlers/test_listeners.py
+++ b/tests/unit/test_handlers/test_websocket_handlers/test_listeners.py
@@ -4,7 +4,7 @@ from typing import Any, AsyncGenerator, Dict, List, Optional, Type, Union, cast
 from unittest.mock import AsyncMock, MagicMock
 
 import pytest
-from pytest_lazyfixture import lazy_fixture
+from pytest_lazy_fixtures import lf
 
 from litestar import Controller, Litestar, Request, WebSocket
 from litestar.datastructures import State
@@ -47,9 +47,9 @@ def async_listener_callable(mock: MagicMock) -> websocket_listener:
 @pytest.mark.parametrize(
     "listener",
     [
-        lazy_fixture("sync_listener_callable"),
-        lazy_fixture("async_listener_callable"),
-        lazy_fixture("listener_class"),
+        lf("sync_listener_callable"),
+        lf("async_listener_callable"),
+        lf("listener_class"),
     ],
 )
 def test_basic_listener(mock: MagicMock, listener: Union[websocket_listener, Type[WebsocketListener]]) -> None:
-- 
2.44.0

